src/App.tsx

```diff
@@ -1,14 +1,66 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { getTodos } from './api';
+import { actions as todosActions } from './features/todos';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+  const status = useAppSelector(state => state.filter.status);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const isSelected = currentTodo !== null;
+
+  const getTodosFromServer = async () => {
+    setIsLoading(true);
+    let allTodos;
+
+    try {
+      allTodos = await getTodos();
+    } catch (error) {
```

Comment:
Don't need to have global `allTodos`. Just wrap all of your code in big `try...catch...` block (Not just await, but also rest of code)